### DISCUSSION TITLE
Don't understand my error (`NotImplementedError`)

### DISCUSSION THREAD

**Discussion by naptax** [mrexodia/dumpulator#51] [Q&A]
HI Dumpulators ;-)

I don't understand my mistake:
```

from dumpulator import Dumpulator

dp = Dumpulator("TryHarder.dmp", quiet=False)
base = 0x13000
stop_address = base + 0x1773

dp.start(base+0x1731, end=stop_address)
```
and a ....  bad **_Not Implemented Error on_**

```
2937 @syscall
   2938 def ZwQueryInformationJobObject(dp: Dumpulator,
   2939                                 JobHandle: Annotated[HANDLE, SAL("_In_opt_")],
   (...)
   2943                                 ReturnLength: Annotated[P(ULONG), SAL("_Out_opt_")]
   2944                                 ):
-> 2945     raise NotImplementedError()

```

Full trace attached.
Any idea about my mistake ?

Thanks for any help.

[logError.txt](https://github.com/mrexodia/dumpulator/files/10842851/logError.txt)

**Comment by mrexodia** [comment]
The cause is quite clear: the function `ZwQueryInformationJobObject` is not implemented in dumpulator. You can implement your own version of the function, see [here](https://github.com/mrexodia/dumpulator/#custom-syscall-implementation) for an example.

Maybe to clarify a bit: dumpulator is just a prototype. It's not a goal to implement all the functions, just the most used ones and then allow the user to stub the functions for their own purposes.

**Reply by naptax** [reply]
Thank you Duncan. I will implement my stubs.
